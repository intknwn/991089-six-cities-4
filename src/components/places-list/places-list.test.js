import React from 'react';
import renderer from 'react-test-renderer';
import PlacesList from './places-list.jsx';

const offers = [{
  picture: `https://via.placeholder.com/260x200?text=Place+1`,
  isPremium: false,
  rate: 230,
  title: `Comfortable Double Room near Amsterdam Center`,
  type: `room`,
  rating: 1,
  isFavorite: false,
  location: [52.3909553943508, 4.85309666406198]
}, {
  picture: `https://via.placeholder.com/260x200?text=Place+2`,
  isPremium: true,
  rate: 150,
  title: `Spacious and stylish downtown suite`,
  type: `house`,
  rating: 5,
  isFavorite: true,
  location: [52.369553943508, 4.85309666406198]
}, {
  picture: `https://via.placeholder.com/260x200?text=Place+3`,
  isPremium: false,
  rate: 110,
  title: `Studio 13`,
  type: `hotel`,
  rating: 2,
  isFavorite: true,
  location: [52.3909553943508, 4.929309666406198]
}, {
  picture: `https://via.placeholder.com/260x200?text=Place+4`,
  isPremium: false,
  rate: 240,
  title: `Cozy room in lively apartment and neighbourhood`,
  type: `apartment`,
  rating: 3,
  isFavorite: false,
  location: [52.3809553943508, 4.939309666406198]
}
];

it(`PlaceCard render`, () => {
  const tree = renderer.create(
      <PlacesList
        places={offers}
        onActiveItemSet={() => {}}
        onCardTitleClick={() => {}}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
