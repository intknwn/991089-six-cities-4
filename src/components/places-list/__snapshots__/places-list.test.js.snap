// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceCard render 1`] = `
Array [
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/1"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/20.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            227
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "40%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/1"
          onClick={[Function]}
        >
          Nice, cozy, warm big bed apartment
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>,
  <article
    className="cities__place-card place-card"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="/offer/2"
        onClick={[Function]}
      >
        <img
          alt="Place image"
          className="place-card__image"
          height={200}
          src="https://htmlacademy-react-3.appspot.com/six-cities/static/hotel/8.jpg"
          width={260}
        />
      </a>
    </div>
    <div
      className="place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            327
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className="place-card__bookmark-button  button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
      >
        <a
          href="/offer/2"
          onClick={[Function]}
        >
          The house among olive
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>,
]
`;
